src/capture/middlewares/header_middleware/dns_decoder.c

```diff
@@ -22,7 +22,7 @@
 void decode_dns_questions(uint8_t *payload, struct capture_packet *cpac) {
   uint16_t idx, i = 0, j = 0;
   for (idx = 0; idx < /*cpac->dnss.nqueries*/ 1; idx++) {
-    while ((((void *)&payload[i] - (void *)cpac->ethh)) < cpac->length) {
+    while ((uint32_t)(((void *)&payload[i] - (void *)cpac->ethh)) < cpac->length) {
```

Comment:
```suggestion
    while ((uint32_t)((&payload[i] - (uint8_t *)cpac->ethh)) < cpac->length) {
```

Doing pointer arithmetic on `void *` is not allowed under the C standard.
(However both GCC and Clang allow it as an extension).

Still, it's worth using `uint8_t *` instead, since it's more standard:
https://gcc.gnu.org/onlinedocs/gcc-10.1.0/gcc/Pointer-Arith.html#Pointer-Arith

Reply:
Makes sense.